### PR TITLE
hw-test: Verify img signature before flashing

### PR DESCRIPTION
**Pull request by henrirosten**
Verify image signature using the tooling from https://github.com/tiiuae/ci-yubi before running any HW-tests

### CODE REVIEW

**Review by cazfi — APPROVED** [review]

**Review by vjuntunen — APPROVED** [review]

File: ghaf-hw-test.groovy [hunk 1]
```diff
@@ -160,6 +161,11 @@ pipeline {
           """
           img_relpath = run_cmd("find ${TMP_IMG_DIR} -type f -print -quit | grep .")
           println "Downloaded image to workspace: ${img_relpath}"
+          // Verify signature using the tooling from: https://github.com/tiiuae/ci-yubi
+          sh "wget -nv -P ${TMP_SIG_DIR} ${params.IMG_URL}.sig"
+          sig_relpath = run_cmd("find ${TMP_SIG_DIR} -type f -print -quit | grep .")
```
Comment (alextserepov) [inline ghaf-hw-test.groovy:166]: please consider adding some input validation to variables used in shell commands.

File: ghaf-hw-test.groovy [hunk 2]
```diff
@@ -160,6 +161,11 @@ pipeline {
           """
           img_relpath = run_cmd("find ${TMP_IMG_DIR} -type f -print -quit | grep .")
           println "Downloaded image to workspace: ${img_relpath}"
+          // Verify signature using the tooling from: https://github.com/tiiuae/ci-yubi
+          sh "wget -nv -P ${TMP_SIG_DIR} ${params.IMG_URL}.sig"
```
Comment (alextserepov) [inline ghaf-hw-test.groovy:165]: Please consider adding IMG_URL validation for making sure we are using TLS (e.g. it starts with https) and also consider shell injection cases.

File: ghaf-hw-test.groovy [hunk 3]
```diff
@@ -160,6 +161,11 @@ pipeline {
           """
           img_relpath = run_cmd("find ${TMP_IMG_DIR} -type f -print -quit | grep .")
```
Comment (alextserepov) [inline ghaf-hw-test.groovy:162]: Please consider possible shell injection cases

File: ghaf-hw-test.groovy [hunk 4]
```diff
@@ -160,6 +161,11 @@ pipeline {
...
           println "Downloaded image to workspace: ${img_relpath}"
+          // Verify signature using the tooling from: https://github.com/tiiuae/ci-yubi
+          sh "wget -nv -P ${TMP_SIG_DIR} ${params.IMG_URL}.sig"
+          sig_relpath = run_cmd("find ${TMP_SIG_DIR} -type f -print -quit | grep .")
+          println "Downloaded signature to workspace: ${sig_relpath}"
+          sh "nix run github:tiiuae/ci-yubi/e2aa4c6#verify -- --path ${img_relpath} --sigfile ${sig_relpath}"
           // Uncompress, keeping only the decompressed image file
           if(img_relpath.endsWith("zst")) {
             sh "zstd -dfv ${img_relpath} && rm ${img_relpath}"
```
Comment (alextserepov) [inline ghaf-hw-test.groovy:171]: same as line 161.

File: ghaf-hw-test.groovy (hunk 1 above)
Comment (henrirosten) [reply ghaf-hw-test.groovy:166]: I'd like handle the input validation in a separate ticket. 
I agree we need to think of how do we handle this properly, across all pipelines not just the one modified here.
Currently, we don't provide any pipelines that take arguments to wider audience.

File: ghaf-hw-test.groovy (hunk 3 above)
Comment (henrirosten) [reply ghaf-hw-test.groovy:162]: See: https://github.com/tiiuae/ghaf-jenkins-pipeline/pull/68#discussion_r1760991470

File: ghaf-hw-test.groovy (hunk 4 above)
Comment (henrirosten) [reply ghaf-hw-test.groovy:171]: See: https://github.com/tiiuae/ghaf-jenkins-pipeline/pull/68#discussion_r1760991470

File: ghaf-hw-test.groovy (hunk 2 above)
Comment (henrirosten) [reply ghaf-hw-test.groovy:165]: See: https://github.com/tiiuae/ghaf-jenkins-pipeline/pull/68#discussion_r1760991470

File: ghaf-hw-test.groovy (hunk 1 above)
Comment (alextserepov) [reply ghaf-hw-test.groovy:166]: sure. Let's take it as a separate ticket.

**Review by alextserepov — APPROVED** [review]

**Pull request merged**